core/src/test/java/org/apache/brooklyn/entity/group/DynamicClusterTest.java

```diff
@@ -1225,4 +1235,93 @@ private void assertFirstAndNonFirstCounts(Collection<Entity> members, int expect
         assertEquals(found.size(), expectedNonFirstCount);
     }
 
+    @DataProvider
+    public Object[][] maxConcurrentCommandsTestProvider() {
+        return new Object[][]{{1}, {2}, {3}};
+    }
+
+    @Test(dataProvider = "maxConcurrentCommandsTestProvider")
+    public void testEntitiesStartAndStopSequentiallyWhenMaxConcurrentCommandsIsOne(int maxConcurrentCommands) {
```

Comment:
I tend to observe high concurrency even for this simple entity. When it fails it's typically with an error like
```
2016-11-28 13:57:33,285 ERROR Cluster DynamicClusterImpl{id=gve8830u85} failed to start entity ThrowOnAsyncStartEntityImpl{id=vofv04muyy} (removing): java.lang.AssertionError: expected 9 <= 1 expected [true] but found [false]
```
And that's with a `finally` block decrementing the counter again.